### PR TITLE
net: remove conn API

### PR DESCRIPTION
**Pull request by miri64**
conn was deprecated in 38217347. 3 Releases later and now that no module
is using it RIOT-internally anymore, I think it is time to say goodbye.

Requires #6974 to pass all tests.

### CODE REVIEW

**Comment by kaspar030** [comment]
>￼ miri64 assigned cgundogan, emmanuelsearch, kaspar030, and smlng 24 minutes ago
￼> miri64 requested review from cgundogan, kaspar030, and smlng 24 minutes ago

We need a "request review from everyone, assign all" button!

**Comment by miri64** [comment]
assignment of multiple people in my case doesn't mean I want all their opinions, just one of them ;-). Those not interested to review can always remove their assignment ;-)

**Comment by miri64** [comment]
Rebased to current master.

File: makefiles/pseudomodules.inc.mk [hunk 1]
```diff
@@ -1,8 +1,5 @@
 PSEUDOMODULES += auto_init_gnrc_rpl
 PSEUDOMODULES += conn
```
Comment (smlng) [inline makefiles/pseudomodules.inc.mk]: should be remove to?

File: pkg/emb6/patches/0004-Provide-event-type-for-RIOT-s-conn_.-_send.patch [hunk 2]
```diff
@@ -23,7 +23,7 @@ index 5316a67..78153be 100644
  #define EVENT_TYPE_ICMP6            0x05    ///< New icmp6 packet event
  #define EVENT_TYPE_TCPIP            0x06    ///< New tcpip event
  #define EVENT_TYPE_SLIP_POLL        0x07    ///< Process slip handler
-+#define EVENT_TYPE_CONN_SEND        0x08    ///< Send command from conn layer
++#define EVENT_TYPE_CONN_SEND        0x08    ///< Send command from sock layer
```
Comment (smlng) [inline pkg/emb6/patches/0004-Provide-event-type-for-RIOT-s-conn_.-_send.patch]: wouldn't `EVENT_TYPE_SOCK_SEND` be more appropriate, though it requires to change every usage of this macro, too.

Comment (smlng) [inline pkg/emb6/patches/0004-Provide-event-type-for-RIOT-s-conn_.-_send.patch]: added in https://github.com/miri64/RIOT/pull/30

**Comment by miri64** [comment]
oops merged

**Comment by smlng** [comment]
@miri64 please squash (including my small additions)

**Comment by smlng** [comment]
and make the merge commit disappear 😉

**Comment by miri64** [comment]
Done

**Comment by smlng** [comment]
a second review and approval would be great!

**Comment by miri64** [comment]
@kaspar030 @emmanuelsearch @cgundogan?

**Comment by smlng** [comment]
ping, anyone willing to ACK, too?

**Comment by kaspar030** [comment]
Let's go!

**Comment by smlng** [comment]
🎉  finally!

**Comment by emmanuelsearch** [comment]
yes!

**Pull request merged**